emcc.py

```diff
@@ -1143,8 +1143,11 @@ def check(input_file):
           # if not wasm-only, we can't do async compilation as the build can run in other
           # modes than wasm (like asm.js) which may not support an async step
           shared.Settings.BINARYEN_ASYNC_COMPILATION = 0
+          warning = 'This will reduce performance and compatibility (some browsers limit synchronous compilation), see https://github.com/kripken/emscripten/wiki/WebAssembly#codegen-effects'
           if 'BINARYEN_ASYNC_COMPILATION=1' in settings_changes:
-            logging.warning('BINARYEN_ASYNC_COMPILATION requested, but disabled since not in wasm-only mode')
+            logging.warning('BINARYEN_ASYNC_COMPILATION requested, but disabled because of user options. ' + warning)
+          elif 'BINARYEN_ASYNC_COMPILATION=0' not in settings_changes:
+            logging.warning('BINARYEN_ASYNC_COMPILATION disabled due to user options. ' + warning)
```

Comment:
Yeah, it would be better to be much more granular. But the link to the wiki page in the error here should help. And I think there will be churn here with options moving to support async compilation, so I'd rather not hardcode a list that could get out of date.

I could be wrong though and we'll see users say this isn't a good enough warning, we can iterate on this then.